airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/logging/sentry.py

```diff
@@ -0,0 +1,226 @@
+    to quiet things down.
+    """
+    from sentry_sdk.integrations.argv import ArgvIntegration
+    from sentry_sdk.integrations.atexit import AtexitIntegration
+    from sentry_sdk.integrations.dedupe import DedupeIntegration
+    from sentry_sdk.integrations.logging import LoggingIntegration, ignore_logger
+    from sentry_sdk.integrations.modules import ModulesIntegration
+    from sentry_sdk.integrations.stdlib import StdlibIntegration
+
+    # We ignore the Dagster internal logging to prevent a single error from being logged per node in the job graph
+    ignore_logger("dagster")
+
+    SENTRY_DSN = os.environ.get("SENTRY_DSN")
+    SENTRY_ENVIRONMENT = os.environ.get("SENTRY_ENVIRONMENT")
+    TRACES_SAMPLE_RATE = float(os.environ.get("SENTRY_TRACES_SAMPLE_RATE", 0))
+
+    sentry_logger.info("Setting up Sentry with")
+    sentry_logger.info(f"SENTRY_DSN: {SENTRY_DSN}")
+    sentry_logger.info(f"SENTRY_ENVIRONMENT: {SENTRY_ENVIRONMENT}")
+    sentry_logger.info(f"SENTRY_TRACES_SAMPLE_RATE: {TRACES_SAMPLE_RATE}")
+
+    if SENTRY_DSN:
+        sentry_sdk.init(
+            dsn=SENTRY_DSN,
+            traces_sample_rate=TRACES_SAMPLE_RATE,
+            environment=SENTRY_ENVIRONMENT,
+            default_integrations=False,
+            integrations=[
+                AtexitIntegration(),
+                DedupeIntegration(),
+                StdlibIntegration(),
+                ModulesIntegration(),
+                ArgvIntegration(),
+                LoggingIntegration(),
+            ],
+        )
+
+
+def _is_context(context):
+    """
+    Check if the given object is a valid context object.
+    """
+    return (
+        isinstance(context, OpExecutionContext)
+        or isinstance(context, SensorEvaluationContext)
+        or isinstance(context, AssetExecutionContext)
+    )
+
+
+def _get_context_from_args_kwargs(args, kwargs):
+    """
+    Given args and kwargs from a function call, return the context object if it exists.
+    """
+    # if the first arg is a context object, return it
+    if len(args) > 0 and _is_context(args[0]):
+        return args[0]
+
+    # if the kwargs contain a context object, return it
+    if "context" in kwargs and _is_context(kwargs["context"]):
+        return kwargs["context"]
+
+    # otherwise raise an error
+    raise Exception(
+        f"No context provided to Sentry Transaction. When using @instrument, ensure that the asset/op has a context as the first argument."
+    )
+
+
+def _with_sentry_op_asset_transaction(context: OpExecutionContext):
+    """
+    Start or continue a Sentry transaction for the Dagster Op/Asset
+    """
+    op_name = context.op_def.name
+    job_name = context.job_name
+
+    sentry_logger.info(f"Initializing Sentry Transaction for Dagster Op/Asset {job_name} - {op_name}")
+    transaction = sentry_sdk.Hub.current.scope.transaction
+    sentry_logger.info(f"Current Sentry Transaction: {transaction}")
```

Comment:
Is this not spammy? Could be a debug log level thing

Reply:
Good call!